### DISCUSSION TITLE
Add Module & Change Entities Migration

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3357] [General]
Hi,

How do I change/prevent the module’s entities / tables creation/migration to the database while adding to the page?

After module added to the page, dynamic form fields will be migrated to the database. How do I do that?

**Comment by sbwalker** [comment]
@Sharafudeen I have no idea what you are trying to do. Typically migrations are executed at startup - if you look at any of the available module examples you will see how migrations are declared based on version#.

**Reply by Sharafudeen** [reply]
@sbwalker How do I run migrations manually using button action in an Oqtane Module? I don’t want to do migrations at startup.

**Reply by sbwalker** [reply]
@Sharafudeen I am not sure if it is possible... as if your model class does not match the schema of your database table, EF will throw an error (which is why migrations are usually run either prior or during startup to ensure the environment is in a stable state).
